src/createServer.js

```diff
@@ -1,3 +1,60 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { convertToCase } = require('./convertToCase');
+
+function createServer() {
+  const server = http.createServer((req, res) => {
+    const [originalText, queryString] = req.url.slice(1).split('?');
+    const targetCase = new URLSearchParams(queryString).get('toCase');
+
+    const errors = [];
+
+    if (!originalText) {
+      errors.push({
+        message: `Text to convert is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".`,
+      });
+    }
+
+    if (!targetCase) {
+      errors.push({
+        message: `"toCase" query param is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".`,
+      });
+    }
+
+    if (
+      targetCase &&
+      !['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'].includes(
+        targetCase.toUpperCase(),
+      )
+    ) {
+      errors.push({
+        message: `This case is not supported. Available cases: SNAKE, KEBAB, CAMEL, PASCAL, UPPER.`,
+      });
+    }
```

Comment:
- Create a function that validates input for errors and move it to a separate file
- Create a variable for `['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER']` and generate the message with "Available cases" depending on this variable